compiler/luci/pass/src/PropagateQParamBackwardPass.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/PropagateQParamBackwardPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/IR/CircleNodeVisitor.h>
+#include <luci/Log.h>
+
+namespace
+{
+
+// Visitor to propagate quantization parameters backwards
+struct PropagateQParamBackward final : public luci::CircleNodeMutableVisitor<void>
+{
+  PropagateQParamBackward(loco::DataType output) : _output_type(output) {}
+
+private:
+  loco::DataType _output_type;
+
+  void visit(luci::CircleNode *) {}
+};
+
+} // namespace
+
+namespace luci
+{
+
+bool PropagateQParamBackwardPass::run(loco::Graph *g)
+{
+  LOGGER(l);
+
+  // We use post-order traversal as qparam is propagated backward
+  for (auto node : loco::postorder_traversal(loco::output_nodes(g)))
+  {
+    auto circle_node = loco::must_cast<luci::CircleNode *>(node);
+    INFO(l) << "PropagateQParamBackwardPass visit node: " << circle_node->name() << std::endl;
+
+    PropagateQParamBackward pqb(_output_model_dtype);
+    circle_node->accept(&pqb);
+  }
+
+  return false;
```

Comment:
After seeing full draft, this pass seems to return always `false`.
I think it is maybe because visitor class return `void` and thus there is nothing to check, am I right?

Then how about changing return type of visitor class and functions to `bool` and then using it to determine whether there are something changed or not?
If this is hard to apply, we may go as it is :D

Reply:
You're right :) The return value of this pass is not used anywhere, so the pass just returns false.

In the following PRs, I'll just move existing codes in QuantizeWithMinMaxPass to the PropagateQParamBackwardPass. For example, `propagate_concat_quantparam`, which returns void. We can modify the existing codes, but it does not give actual benefit because the return value is not used.

Another option would be implementing `PropagateQParamBackward` without inheriting `logo::Pass`, which looks better to me. I used `logo::Pass` for consistency with `PropagateQParamForwardPass`, but I'm very open to change the decision.

Reply:
> We can modify the existing codes, but it does not give actual benefit because the return value is not used.

I agree with it :)

> implementing PropagateQParamBackward without inheriting logo::Pass

It looks better to me either :D
But for now, we may go as it is and then discuss later how to refactoring this case!
If another similar passes are created, it would be more helpful to think which solution would be better :)
Could you leave comments like following example?

```
  // This pass is only run once, so return false
  // TODO Refactoring not to return meaningless value
```

Reply:
Not for now but maybe good solution :D
We can introduce new phase such as `logo::PhaseStrategy::OnlyOnce` for these cases :)